libs/libdaemon/src/BaseDaemon.cpp

```diff
@@ -921,11 +922,35 @@ void BaseDaemon::defineOptions(Poco::Util::OptionSet& _options)
             );
 }
 
+bool is_pid_running(pid_t pid)
+{
+    if (0 == kill(pid, 0))
```

Comment:
> If  sig  is  0,  then  no signal is sent, but existence **and permission checks** are still performed; this can be used to check for the existence of a process ID or process group ID that the caller **is permitted to signal**.

Excessive check. There is the possibility that the process is running from different user.